### PR TITLE
Feature/juan/108/new subscriptions

### PR DESCRIPTION
**Pull request by justinian336**
Implements the new subscriptions system

### CODE REVIEW

File: src/main/java/jp/gopay/sdk/resources/SubscriptionsResource.java [hunk 1]
```diff
@@ -35,38 +40,69 @@
...
+    Call<FullSubscription> updateSubscription(
             @Path("storeId") StoreId storeId,
             @Path("subscriptionId") SubscriptionId subscriptionId,
-            @Body SubscriptionPatchData dataToPost,
+            @Body SubscriptionUpdateData dataToPost,
             @Header(idempotencyKeyHeaderName) IdempotencyKey idempotencyKey
     );
 
-    @GET("/stores/{storeId}/subscriptions/{subscriptionId}/charges")
-    Call<PaginatedList<Charge>> listSubscriptionCharges(
```
Comment (shiraki-y) [inline src/main/java/jp/gopay/sdk/resources/SubscriptionsResource.java]: This route is deleted and for now there is no same route in java sdk. Will it be added later or should it be deleted for any reason?

File: src/main/java/jp/gopay/sdk/resources/SubscriptionsResource.java [hunk 2]
```diff
@@ -35,38 +40,69 @@
...
     @DELETE("/stores/{storeId}/subscriptions/{subscriptionId}")
     Call<Void> deleteSubscription(
             @Path("storeId") StoreId storeId,
             @Path("subscriptionId") SubscriptionId subscriptionId
     );
+
+    @POST("/subscriptions/simulate_plan")
+    Call<PaymentsPlan> simulateSubscriptionPlan(
+            @Body SubscriptionPlanSimulationRequest simulationRequest
+    );
```
Comment (shiraki-y) [inline src/main/java/jp/gopay/sdk/resources/SubscriptionsResource.java]: `POST          /stores/:storeId/subscriptions/simulate_plan`
Isn’t this route necessary for now?

File: src/main/java/jp/gopay/sdk/models/request/subscription/ScheduledPaymentPatchData.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package jp.gopay.sdk.models.request.subscription;
+
+import com.google.gson.annotations.SerializedName;
+import org.joda.time.LocalDate;
+
+public class ScheduledPaymentPatchData {
+
+    @SerializedName("due_date")
+    private LocalDate dueDate;
```
Comment (shiraki-y) [inline src/main/java/jp/gopay/sdk/models/request/subscription/ScheduledPaymentPatchData.java]: Is patching `due-date` allowed only the admin?

File: src/main/java/jp/gopay/sdk/models/response/subscription/ScheduleSettings.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    @SerializedName("start_on")
+    private LocalDate startOn;
+
+    @SerializedName("zone_id")
+    private ZoneId zoneId;
+
+    @SerializedName("preserve_end_of_month")
+    private Boolean preserveEndOfMonth;
+
+    public ScheduleSettings(LocalDate startOn, ZoneId zoneId, Boolean preserveEndOfMonth) {
```
Comment (shiraki-y) [inline src/main/java/jp/gopay/sdk/models/response/subscription/ScheduleSettings.java]: Isn’t `startDateReference` necessary?

File: src/main/java/jp/gopay/sdk/models/response/subscription/FullSubscription.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    @SerializedName("installment_plan")
+    private InstallmentPlan installmentPlan;
+
+    @SerializedName("next_payment")
+    private ScheduledPayment nextPayment;
+
+    @SerializedName("payments_left")
+    private Integer paymentsLeft;
+
```
Comment (shiraki-y) [inline src/main/java/jp/gopay/sdk/models/response/subscription/FullSubscription.java]: Aren’t `amountLeft` and `amountLeftFormatted` necessary?

File: src/main/java/jp/gopay/sdk/resources/SubscriptionsResource.java (hunk 1 above)
Comment (justinian336) [reply src/main/java/jp/gopay/sdk/resources/SubscriptionsResource.java]: For a moment I thought that this route had been deleted in the API, but checking again it's still there. I will include it again.

File: src/main/java/jp/gopay/sdk/resources/SubscriptionsResource.java (hunk 2 above)
Comment (justinian336) [reply src/main/java/jp/gopay/sdk/resources/SubscriptionsResource.java]: Yes, it should be included. I'll add it

File: src/main/java/jp/gopay/sdk/models/request/subscription/ScheduledPaymentPatchData.java (hunk 3 above)
Comment (justinian336) [reply src/main/java/jp/gopay/sdk/models/request/subscription/ScheduledPaymentPatchData.java]: You're right.

File: src/main/java/jp/gopay/sdk/models/response/subscription/ScheduleSettings.java (hunk 4 above)
Comment (justinian336) [reply src/main/java/jp/gopay/sdk/models/response/subscription/ScheduleSettings.java]: `startDateReference` is not in the API response.

File: src/main/java/jp/gopay/sdk/models/response/subscription/FullSubscription.java (hunk 5 above)
Comment (justinian336) [reply src/main/java/jp/gopay/sdk/models/response/subscription/FullSubscription.java]: Thank you, I hadn't seen that. I'll fix it.

**Comment by justinian336** [comment]
Ok, everything fixed and the integration tests are green. I'll proceed to merge.

**Pull request merged**